.scripts/get-interface-test-targets.py

```diff
@@ -94,17 +99,27 @@ def _target_from_interface(
             - endpoint (name)
     """
     interface = _REPO_ROOT / interface_str
+    version_dirs = sorted((interface / 'interface').glob('v[0-9]*'))
+    if not version_dirs:
+        logger.warning('%s does not define any interface versions.', interface_str)
+        return []
     targets: list[dict[str, str]] = []
-    for v in sorted((interface / 'interface').glob('v[0-9]*')):
+    for v in version_dirs:
         interface_yaml = yaml.safe_load((v / 'interface.yaml').read_text())
+        if (disable := v / 'tests' / '.disable').exists():
+            msg = 'Targets for %s %s will not be included since %s exists.'
+            logger.warning(msg, interface_str, v.name, disable)
+            continue
         for role in 'provide', 'require':
             charms = interface_yaml.get(f'{role}rs', [])
-            if not charms:
-                continue
             if has_tests_only and not _has_tests(v, f'{role}r'):
                 msg = 'Skipping these charms because there are no tests for %s %s %s: %s'
                 logger.warning(msg, interface_str, v.name, role, charms)
                 continue
+            if not charms:
+                msg = 'No charms for %s %s %s'
+                logger.warning(msg, interface_str, v.name, role)
```

Comment:
Any reason not to not just do this?


```suggestion
                logger.warning('No charms for %s %s %s', interface_str, v.name, role)
```